crates/hotshot/examples/infra/mod.rs

```diff
@@ -872,20 +877,30 @@ pub async fn main_entry_point<
 
     // conditionally save/load config from file or orchestrator
     let (mut run_config, source) =
-        NetworkConfig::from_file_or_orchestrator(&orchestrator_client, args.network_config_file)
-            .await;
+        NetworkConfig::<TYPES::SignatureKey, TYPES::ElectionConfigType>::from_file_or_orchestrator(
+            &orchestrator_client,
+            args.clone().network_config_file,
+        )
+        .await;
 
     let node_index = run_config.node_index;
     error!("Retrieved config; our node index is {node_index}");
 
-    run_config.config.my_own_validator_config =
-        ValidatorConfig::<<TYPES as NodeType>::SignatureKey>::generated_from_seed_indexed(
-            run_config.seed,
-            node_index,
-            1,
-        );
     //run_config.libp2p_config.as_mut().unwrap().public_ip = args.public_ip.unwrap();
 
+    // one more round of orchestrator here to get peer's public key/config
+    orchestrator_client
+        .post_and_wait_all_public_keys::<TYPES::SignatureKey, TYPES::ElectionConfigType>(
```

Comment:
Why cant we return the updated peer config here?

Reply:
Previously I separated them due to a weird bug which turned out not to be a bug at all lol, now they're merged to one function.

Reply:
great!